consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/LocalNodeIsProposerTest.java

```diff
@@ -0,0 +1,116 @@
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.CommitPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.ProposalPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContextFactory;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import java.time.Clock;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class LocalNodeIsProposerTest {
+  // These tests assume the basic function of the Ibft Round State Machine has been proven
+  // via the LocalNodeIsNotProposerTest.
+
+  private final long blockTimeStamp = 100;
+  private final Clock mockBlockCreationTime = mock(Clock.class);
+
+  // Local node will propose the first block
+  private final TestContext context =
+      TestContextFactory.createTestEnvironment(4, 1, mockBlockCreationTime);
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private Block expectedProposedBlock;
+  private SignedData<ProposalPayload> expectedTxProposal;
+  private SignedData<CommitPayload> expectedTxCommit;
+
+  @Before
+  public void setup() {
+    when(mockBlockCreationTime.millis()).thenReturn(blockTimeStamp * 1000);
+    expectedProposedBlock = context.createBlockForProposal(0, blockTimeStamp);
+    expectedTxProposal =
+        localNodeMessageFactory.createSignedProposalPayload(roundId, expectedProposedBlock);
+
+    final IbftExtraData extraData =
+        IbftExtraData.decode(expectedProposedBlock.getHeader().getExtraData());
+    final Signature commitSeal =
+        SECP256K1.sign(
+            IbftBlockHashing.calculateDataHashForCommittedSeal(
+                expectedProposedBlock.getHeader(), extraData),
+            context.getLocalNodeParams().getNodeKeyPair());
+
+    expectedTxCommit =
+        localNodeMessageFactory.createSignedCommitPayload(
+            roundId, expectedProposedBlock.getHash(), commitSeal);
+
+    // Start the Controller, and trigger "block timer" to send proposal.
+    context.getController().start();
+    context.getController().handleBlockTimerExpiry(new BlockTimerExpiry(roundId));
+  }
+
+  @Test
+  public void basicCase() {
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxProposal);
+
+    // NOTE: In these test roles.getProposer() will return NULL.
+    roles.getNonProposingPeers().get(0).injectPrepare(roundId, expectedProposedBlock.getHash());
```

Comment:
nit; for a readability gain, if you changed the signature of `getNonProposingPeers()`to accept an integer then `getNonProposingPeers().get(0)` could become `getNonProposingPeer(0)`.

Reply:
yup - much nicer solution.